docs/dev/Pagination-v2.md

```diff
@@ -285,3 +285,87 @@ OpenSearchExecutionEngine->>+ProjectOperator: getTotalHits
     ResourceMonitorPlan-->>-ProjectOperator: value
   ProjectOperator-->>-OpenSearchExecutionEngine: value
 ```
+
+### Plan Tree changes
+
+#### Abstract Plan tree
+
+Changes:
+1. New Plan node -- `Paginate` -- added into the tree.
+2. `QueryPlan` got new optional field: page size. When it is set, `Paginate` is being added. It is converted to `LogicalPaginate` later. For non-paging requests the tree remains unchanged.
+
+```mermaid
+classDiagram
+  direction LR
+  class QueryPlan {
+    <<AbstractPlan>>
+    -int pageSize
+  }
+  class Paginate {
+    <<UnresolvedPlan>>
+  }
+  class UnresolvedPlanTree {
+    <<UnresolvedPlan>>
+  }
+  QueryPlan --* Paginate
+  Paginate --* UnresolvedPlanTree
+```
+#### Logical Plan tree
+
+Changes:
+1. `LogicalPaginate` is added to the top of the tree. It has a private field `pageSize` being pushed down in the `Optimizer`.
```

Comment:
Following your template from above, consider adding: 
2. How is this different from a non-pagination logical plan?

Reply:
Also include: what is the responsibility of the LogicalPaginate class for the LogicalPlan